File: ios/Sources/ACActions/PopoverActionHandler.swift
// Copyright (c) Microsoft Corporation. All rights reserved.
// Author: Vikrant Singh (github.com/VikrantSingh01)
// Licensed under the MIT License.

import SwiftUI
import ACCore

@MainActor
public class PopoverActionHandler {
    public static func handle(action: PopoverAction, delegate: ActionDelegate?) {
        // TODO: Add appropriate delegate method for PopoverAction
        print("PopoverAction triggered: \(action)")
    }
}

File: android/ac-actions/src/main/kotlin/com/microsoft/adaptivecards/actions/PopoverActionHandler.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Author: Vikrant Singh (github.com/VikrantSingh01)
// Licensed under the MIT License.

package com.microsoft.adaptivecards.actions

import com.microsoft.adaptivecards.core.models.ActionPopover

object PopoverActionHandler {
    fun handle(action: ActionPopover, delegate: ActionDelegate?) {
        // Popover actions are UI-driven; delegate notification only
        delegate?.onShowCard(action.id ?: "", true)
    }
}
